### PR TITLE
Remove use of `optin_builtin_traits` feature.

### PR DESCRIPTION
**Pull request by jonas-schievink**
Instead of `impl !Send for Threshold`, put a phantom raw pointer in
there and impl `Sync` manually (I presume this being `Sync` was
intended?).

part of https://github.com/japaric/cortex-m-rtfm/issues/22

### CODE REVIEW

**Comment by japaric** [comment]
Thanks, @jonas-schievink.

I was wondering what was the replacement of the NoSendMarker (or whatever they were called) since they got removed a while ago but this makes sense.

> I presume this being Sync was intended?

No, actually I don't think `Threshold` should be `Sync` though I don't see how it being `Sync` could cause any trouble (memory unsafety) right now given that the only channel like mechanism requires 'static lifetime and you'll only ever encounter non static `&[mut] Threshold` references in safe code. Still I think it's better to make it non-Sync.

TL;DR I'll merge this with the `unsafe impl Sync` removed
